api/src/services/survey-critter-service.ts

```diff
@@ -249,34 +249,36 @@ export class SurveyCritterService extends DBService {
   }
 
   /**
-   * Get mapping of `alias` (animal_id) -> critterbase `critter_id` (UUID) for a survey.
-   * For a survey the critter alias works as a unique identifier of a Critterbase critter.
+   * Get mapping of `alias` (animal_id) -> critterbase critter for a survey.
    *
+   * Note: Aliases are converted to lowercase, when accessing the value use .toLowerCase()
    * Note: Business expects unique critter alias' in Surveys effective 01/06/2024
    *
    * @async
    * @param {number} surveyId
-   * @returns {Promise<Map<string, string | undefined>>} Critter alias -> critter_id (UUID) Map
+   * @returns {Promise<Map<string, ICritterDetailed | undefined>>} Critter alias -> Detailed critter
    */
-  async getSurveyCritterIdAliasMap(surveyId: number): Promise<Map<string, string | undefined>> {
+  async getSurveyCritterIdAliasMap(surveyId: number): Promise<Map<string, ICritterDetailed | undefined>> {
     const critters = await this.getCritterbaseSurveyCritters(surveyId);
 
     // Create mapping of alias -> critter_id
     const critterAliasMap = new Map();
-    for (const critter of critters) {
-      // Do not allow existing duplicate aliases to map over eachother
-      if (critterAliasMap.get(critter.animal_id)) {
-        defaultLog.debug({
-          label: 'critter alias map',
-          message: 'duplicate critter alias for survey',
-          details: { surveyId: surveyId, alias: critter.animal_id }
-        });
-
-        critterAliasMap.set(critter.animal_id, undefined);
-      }
 
+    for (const critter of critters) {
       if (critter.animal_id) {
-        critterAliasMap.set(critter.animal_id, critter.critter_id);
+        const alias = critter.animal_id.toLowerCase();
+        // Do not allow existing duplicate aliases to map over eachother
+        if (critterAliasMap.get(alias)) {
+          defaultLog.debug({
+            label: 'critter alias map',
+            message: 'duplicate critter alias for survey',
+            details: { surveyId: surveyId, critter: critter }
+          });
+
+          critterAliasMap.set(alias, undefined);
```

Comment:
Is this how you trigger the zod error? Maybe worth adding a comment just in case

Reply:
I guess the log above kind of describes the situation, though a separate comment here might still be safest

Reply:
updated